src/OctopusPuppet.Gui/ViewModels/DeploymentPlannerViewModel.cs

```diff
@@ -657,6 +657,157 @@ public void LoadEnvironmentDeploymentJson()
+                NotifyOfPropertyChange(() => SelectedSetAllDeploymentsTo);
+            }
+        }
+
+        public void SetAllDeploymentsTo()
+        {
+            if (EnvironmentDeployment == null)
+            {
+                return;
+            }
+
+            var componentsToDeploy = EnvironmentDeployment
+                .ProductDeployments
+                .SelectMany(x => x.DeploymentSteps)
+                .SelectMany(x => x.ComponentDeployments)
+                .Where(x => x.Vertex.DeploymentAction != SelectedSetAllDeploymentsTo)
+                .ToList();
+
+            var deploymentPlans = EnvironmentDeploymentPlan.DeploymentPlans
+                .Where(x => x.Action != SelectedSetAllDeploymentsTo)
+                .ToList();
+
+            foreach (var componentToDeploy in componentsToDeploy)
+            {
+                componentToDeploy.Vertex.DeploymentAction = SelectedSetAllDeploymentsTo;
+            }
+
+            foreach (var deploymentPlan in deploymentPlans)
+            {
+                deploymentPlan.Action = SelectedSetAllDeploymentsTo;
+            }
+
+            if (componentsToDeploy.Any())
+            {
+                var temp = EnvironmentDeployment;
+                EnvironmentDeployment = null;
+                EnvironmentDeployment = temp;
+            }
+
+            if (deploymentPlans.Any())
+            {
+                var temp = EnvironmentDeploymentPlan;
+                EnvironmentDeploymentPlan = null;
+                EnvironmentDeploymentPlan = temp;
+            }
+        }
+
+        public void SkipAllPassedDeployments()
+        {
+            var successfullyDeployedComponents = EnvironmentDeploymentResult
+                .ProductDeployments
+                .SelectMany(x => x.DeploymentSteps)
+                .SelectMany(x => x.ComponentDeployments)
+                .Where(x => x.Status == ComponentVertexDeploymentStatus.Success && x.Vertex.DeploymentAction != PlanAction.Skip)
+                .ToList();
+
+            var componentsToDeploy = EnvironmentDeployment
+                .ProductDeployments
+                .SelectMany(x => x.DeploymentSteps)
+                .SelectMany(x => x.ComponentDeployments)
+                .Where(x => successfullyDeployedComponents.Any(y => y.Vertex.Equals(x.Vertex)))
+                .ToList();
+
+            var deploymentPlans = EnvironmentDeploymentPlan.DeploymentPlans
+                .Where(x => successfullyDeployedComponents.Any(y => y.Vertex.Id == x.Id && y.Vertex.Name == x.Name))
+                .ToList();
+
+            foreach (var successfullyDeployedComponent in successfullyDeployedComponents)
+            {
+                successfullyDeployedComponent.Vertex.DeploymentAction = PlanAction.Skip;
+            }
+
+            foreach (var componentToDeploy in componentsToDeploy)
+            {
+                componentToDeploy.Vertex.DeploymentAction = PlanAction.Skip;
```

Comment:
I thought this was the the one above 'successfullyDeployedComponent'. And was expecting 'componentToDeploy' to be set as to deploy.